browser/ui/webui/brave_rewards_page_ui.cc

```diff
@@ -627,15 +634,15 @@ void RewardsDOMHandler::OnJavascriptAllowed() {
     rewards_service_observation_.Observe(rewards_service_);
   }
 
-  if (ads_service_) {
-    ads_service_observation_.Reset();
-    ads_service_observation_.Observe(ads_service_);
-  }
+  bat_ads_observer_receiver_.reset();
+  ads_service_->AddBatAdsObserver(
```

Comment:
check condition if ads_service_ is not null was deleted

Reply:
Fixed